llvm/lib/Transforms/InstCombine/InstCombineInternal.h

```diff
@@ -785,6 +785,18 @@ class LLVM_LIBRARY_VISIBILITY InstCombinerImpl final
   void handlePotentiallyDeadBlocks(SmallVectorImpl<BasicBlock *> &Worklist);
   void handlePotentiallyDeadSuccessors(BasicBlock *BB, BasicBlock *LiveSucc);
   void freelyInvertAllUsersOf(Value *V, Value *IgnoredUser = nullptr);
+
+  // Take the exact integer log2 of the value. If DoFold is true, create the
+  // actual instructions, otherwise return a non-null dummy value. Return
+  // nullptr on failure. Note, if DoFold is true the caller must ensure that
+  // takeLog2 will succeed, otherwise it may create stray instructions.
```

Comment:
```suggestion
  /// Take the exact integer log2 of the value. If DoFold is true, create the
  /// actual instructions, otherwise return a non-null dummy value. Return
  /// nullptr on failure. Note, if DoFold is true the caller must ensure that
  /// takeLog2 will succeed, otherwise it may create stray instructions.
```